app/commands/user_track/generate_exercise_status_mapping.rb

```diff
@@ -0,0 +1,48 @@
+class UserTrack
+  class GenerateExerciseStatusMapping
+    include Mandate
+
+    initialize_with :track, :user_track
+
+    def call
+      concept_slugs = track.concepts.pluck(:slug)
+      concept_slugs.each.with_object({}) do |slug, hash|
+        hash[slug] = mapping[slug]
+      end
+    end
+
+    private
+    memoize
+    def mapping
+      mapping = Hash.new { |k, v| k[v] = Set.new }
+
+      Exercise::TaughtConcept.joins(:exercise, :concept).
+        where('exercises.track_id': track.id).
+        pluck("track_concepts.slug", "exercises.slug").
+        each do |concept_slug, exercise_slug|
+        next if concept_slug.nil?
+        next if exercise_slug.nil?
+
+        mapping[concept_slug] << exercise_slug
+      end
+
+      # TOOD: Change to practice link and remove limit
+      Exercise::Prerequisite.joins(:exercise, :concept).
```

Comment:
@neenjaw Once you rebase, you can change this to `Exercise::PracticedConcept` and remove the `limit` two lines below.

Reply:
done